android/engine/src/main/java/org/smartregister/fhircore/engine/rulesengine/RulesFactory.kt

```diff
@@ -420,24 +421,31 @@ constructor(
     }
 
     fun generateTaskServiceStatus(task: Task): String {
-      return when (task.status) {
-        Task.TaskStatus.NULL,
-        Task.TaskStatus.FAILED,
-        Task.TaskStatus.RECEIVED,
-        Task.TaskStatus.ENTEREDINERROR,
-        Task.TaskStatus.ACCEPTED,
-        Task.TaskStatus.REJECTED,
-        Task.TaskStatus.DRAFT,
-        Task.TaskStatus.ONHOLD -> {
-          Timber.e("Task.status is null", Exception())
-          ServiceStatus.DUE.name
-        }
-        Task.TaskStatus.REQUESTED -> ServiceStatus.UPCOMING.name
-        Task.TaskStatus.READY -> ServiceStatus.DUE.name
-        Task.TaskStatus.CANCELLED -> ServiceStatus.EXPIRED.name
-        Task.TaskStatus.INPROGRESS -> ServiceStatus.IN_PROGRESS.name
-        Task.TaskStatus.COMPLETED -> ServiceStatus.COMPLETED.name
+      val serviceStatus: String
+      if (task.isOverDue()) {
+        serviceStatus = ServiceStatus.OVERDUE.name
+      } else {
+        serviceStatus =
+          when (task.status) {
+            Task.TaskStatus.NULL,
+            Task.TaskStatus.FAILED,
+            Task.TaskStatus.RECEIVED,
+            Task.TaskStatus.ENTEREDINERROR,
+            Task.TaskStatus.ACCEPTED,
+            Task.TaskStatus.REJECTED,
+            Task.TaskStatus.DRAFT,
+            Task.TaskStatus.ONHOLD -> {
+              Timber.e("Task.status is null", Exception())
+              ServiceStatus.DUE.name
```

Comment:
I do not think we need to use the DUE state here. It's already used when task status is READY.

Reply:
Makes sense. I think I will make them UPCOMING. Since it seems like the task is not ready to be worked on if it's on hold